internal/gatewayapi/translator.go

```diff
@@ -279,6 +282,29 @@ func (t *Translator) GetRelevantGateways(gateways []*gwapiv1.Gateway) []*Gateway
 	return relevant
 }
 
+func (t *Translator) attachEnvoyProxy(gateway *GatewayContext, resources *Resources) {
+	if gateway.Spec.Infrastructure != nil && gateway.Spec.Infrastructure.ParametersRef != nil {
+		if t.MergeGateways {
+			gateway.envoyProxy = resources.ClassEnvoyProxy
+			return
+		}
+
+		ref := gateway.Spec.Infrastructure.ParametersRef
```

Comment:
can we move this into a helper like https://github.com/envoyproxy/gateway/blob/33fceb05ca611e30107ff841cac9d7762eea2048/internal/gatewayapi/resource.go#L97